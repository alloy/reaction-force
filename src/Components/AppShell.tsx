import { Box } from "@artsy/palette"
import { NetworkOfflineMonitor } from "Artsy/Router/NetworkOfflineMonitor"
import { findCurrentRoute } from "Artsy/Router/Utils/findCurrentRoute"
import { useMaybeReloadAfterInquirySignIn } from "Artsy/Router/Utils/useMaybeReloadAfterInquirySignIn"
import { Match } from "found"
import { isFunction } from "lodash"
import React, { useEffect } from "react"
import createLogger from "Utils/logger"

const logger = createLogger("Components/AppShell")

interface AppShellProps {
  children: React.ReactNode
  match: Match
}

export const AppShell: React.FC<AppShellProps> = props => {
  const { children, match } = props
  const routeConfig = findCurrentRoute(match)

  /**
   * Check to see if a route has a prepare key; if so call it. Used typically to
   * preload bundle-split components (import()) while the route is fetching data
   * in the background.
   */
  useEffect(() => {
    if (isFunction(routeConfig.prepare)) {
      try {
        routeConfig.prepare()
      } catch (error) {
        logger.error(error)
      }
    }
  }, [routeConfig])

  /**
   * Let our end-to-end tests know that the app is hydrated and ready to go
   */
  useEffect(() => {
    document.body.setAttribute("data-test", "AppReady")
  }, [])

  // TODO: When old backbone inquiry modal goes away, this can be removed
  useMaybeReloadAfterInquirySignIn()

  return (
    <Box width="100%">
      <Box>
        <Box>{children}</Box>
      </Box>

      <NetworkOfflineMonitor />
    </Box>
  )
}
