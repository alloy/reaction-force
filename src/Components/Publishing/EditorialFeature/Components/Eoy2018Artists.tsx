import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Artists: React.SFC<ArticleProps> = props => {
  return (
    <ArticleWrapper>
      <Nav canFix={false} transparent />
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  background: blue;
  color: white;
  padding-top: 75px;

  .content-end {
    background: white;
  }

  div[class^="FeatureInnerContent__"] {
    color: white;
  }
  div[class^="Author__"] {
    color: white;
    &:before {
      background-color: white;
    }
  }
  .Byline {
    color: white;
    svg path {
      fill: white;
    }
  }
`
