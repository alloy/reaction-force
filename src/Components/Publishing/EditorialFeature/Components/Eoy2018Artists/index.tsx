import { color, Flex, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import { ImageSetPreview } from "Components/Publishing/Sections/ImageSetPreview"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<ArticleProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url

    return (
      <ArtistHeaderSection key={i}>
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg src={src && resize(src, { width: 700 })} />
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    return (
      <TextSection size="5" key={i}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </TextSection>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <React.Fragment>
        <Nav canFix color={color("black100")} backgroundColor="white" />
        <ArticleWrapper>
          <Eoy2018ArticleHeader images={headerImages} />
          <ArticleContent>
            <IntroSection alignItems="flex-start" pl={20}>
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </ArticleWrapper>
      </React.Fragment>
    )
  }
}

const BORDER_WIDTH = 6

const ArticleWrapper = styled.div`
  padding: 0 40px;
`

const ArticleContent = styled.div`
  padding: 40px 0;
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
  }
`

const ArtistHeaderSection = styled.div`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
  display: flex;
  margin-bottom: 40px;
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: 20px;
    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }
`

const ArtistHeaderImg = styled.div<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};
  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
  }
`

const TextSection = styled(Serif)`
  max-width: 75%;
  margin-left: auto;
  padding-bottom: 60px;

  p {
    font-size: 24px;
    text-indent: 2em;

    &:first-child {
      text-indent: 0;
    }
  }
`
const ImageSetWrapper = styled.div`
  margin-bottom: 60px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
`
