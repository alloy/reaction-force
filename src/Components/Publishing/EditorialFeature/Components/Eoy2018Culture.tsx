import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
  return (
    <ArticleWrapper>
      <Nav canFix={false} transparent />
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  background: yellow;
  padding-top: 75px;
`
