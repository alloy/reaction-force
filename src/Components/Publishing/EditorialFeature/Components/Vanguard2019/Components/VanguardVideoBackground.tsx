import { Box } from "@artsy/palette"
import React from "react"
import styled from "styled-components"

interface SVGVideoBackgroundProps {
  id?: string
  url?: string
}

export const VanguardVideoBackground = props => (
  <VideoBackgroundWrapper>
    <svg
      height="0"
      width="0"
      viewBox="0 0 1600 900"
      xmlns="http://www.w3.org/2000/svg"
      preserveAspectRatio="xMinYMin meet"
      clipPathUnits="objectBoundingBox"
      {...props}
    >
      <defs>{getSVGClipPath(props.id)}</defs>
    </svg>
    <VanguardVideo {...props} />
  </VideoBackgroundWrapper>
)

const getSVGClipPath = id => {
  switch (id) {
    case "#clip-svg-intro":
      return (
        <clipPath id={id.slice(1)}>
          <path d="M504.08 63.72c35.91 1.64 149-21.28 207.51-39.9s176.91-16 248.74 0 186.22-1.33 262 1.33 275.35 5.32 283.33 234.11-20 235.44 0 359.14-66.51 205.92-388.41 240.76-538.72 61.19-699.67-12-244.75-34.58-297.95-94.44-58.49-131.66-71.79-186.19-50.5-192.43 13.47-323.81c40.62-83.44 50.38-148.4 146.69-183 72.79-26.07 208.29.01 296.08 4z" />
        </clipPath>
      )
    case "#clip-svg-emerging":
      return (
        <clipPath id={id.slice(1)}>
          <path d="M319.96 4.17L47.2 182.84l99.41 622.66 1343.22 90.93 64.26-658.28-360.06-147.99L319.96 4.17z" />
        </clipPath>
      )
    case "#clip-svg-getting-their-due":
      return (
        <clipPath id={id.slice(1)}>
          <path d="M1488.94 4L65.88 87.92l47.2 800.35 1423.07-83.92L1488.96 4h-.02z" />
        </clipPath>
      )
    case "#clip-svg-newly-established":
      return (
        <clipPath id={id.slice(1)}>
          <path d="M103.9 84.75a631.43 631.43 0 0195.48-21.84c56.68-8.43 113.86-9.53 171-12.3q72.95-2.7 145.49-11.07Q583.17 30.78 650.6 23c51.28-5.2 103.25-9.27 154.82-9.33 56-.05 111.88 5.93 167.63 10.71 67.17 5.76 134.18 35.8 201.61 37.05 23.86 0 47.71-13.15 71.56-12.31 25.79 1.88 51-4.07 75.48 4.34 40.51 12.77 80.73 30.95 117.85 51.75 22.73 12.74 43.63 26.54 60.68 46.63 13.57 16 24.59 34.74 35.61 52.58q11.06 18.33 21.74 36.87a199.56 199.56 0 0120.92 42.12c14.72 45.82 2.3 93-5 138.87-9.83 62.06-6.47 128.19-29 187.64a281.32 281.32 0 01-49.61 82.79l-32.41 30.52c-33.51 26.21-58 60-96.63 77.77-57.73 25.72-133.38 57.1-195.88 63.95q-60.6 4.67-121.34 6.67-76.32 6.38-152.58 13.26c-65.83 5-132.21 10.43-198.11 3.31-61.12-6.6-122.78-19.64-181.87-36.52C468 837.92 420.34 823 377.46 796.45q-40.68-28.23-81.08-56.86-35-24-70.7-47c-23.26-14.9-50.81-27.78-71-47.18-18.86-18.11-34.77-73-46.11-96.52l-10.55-26.38q-12-34-24.84-67.8a353.7 353.7 0 01-17.6-101.19C54 302.39 58.53 250 64.36 199.22c2.28-19.87 5-40.17 10.39-59.47 4.08-14.44 11.47-26.14 17.48-39.24.82-1.78 1.62-3.59 2.38-5.44 0 0 2.19-8.1 9.29-10.32" />
        </clipPath>
      )
  }
}

const VanguardVideo = props => (
  <VideoWrapper>
    <Video
      autoPlay
      loop
      muted
      playsInline
      controls={false}
      src={props.url}
      {...props}
    />
  </VideoWrapper>
)

const VideoWrapper = styled(Box)`
  height: 100vh;
  width: 100vw;
  overflow: hidden;
`
const VideoBackgroundWrapper = styled(Box)`
  height: 100vh;
  width: 100vw;
  position: absolute;
  overflow-x: hidden;
`
const Video = styled.video<SVGVideoBackgroundProps>`
  z-index: -1;
  clip-path: url(${props => props.id});
`
