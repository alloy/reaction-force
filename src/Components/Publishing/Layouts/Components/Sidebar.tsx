import React from "react"
import styled from "styled-components"
import { pMedia } from "../../../Helpers"
import { RelatedArticlesPanel } from "../../RelatedArticles/Panel/RelatedArticlesPanel"

export interface SidebarProps {
  DisplayPanel?: any
  relatedArticlesForPanel?: any
}

export const Sidebar: React.SFC<SidebarProps> = props => {
  const { DisplayPanel, relatedArticlesForPanel } = props

  return (
    <SidebarContainer>
      {relatedArticlesForPanel && (
        <SidebarItem>
          <RelatedArticlesPanel
            label={"Related Stories"}
            articles={relatedArticlesForPanel}
          />
        </SidebarItem>
      )}

      {DisplayPanel && <DisplayPanel />}
    </SidebarContainer>
  )
}

const SidebarContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: -58px 0 0 60px;
  min-width: 280px;
  ${pMedia.xl`
    margin-left: 40px;
  `};
  ${pMedia.lg`
    display: none;
  `};
`

const SidebarItem = styled.div`
  margin-bottom: 40px;
`
