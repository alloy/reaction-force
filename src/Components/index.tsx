import Artwork from "./Artwork"
import ArtworkFilter from "./ArtworkFilter"
import ArtworkGrid from "./ArtworkGrid"
import BorderedPulldown from "./BorderedPulldown"
import Buttons from "./Buttons"
import Checkbox from "./Checkbox"
import Gene from "./Gene"
import Grid from "./Grid"
import Icon from "./Icon"
import Input from "./Input"
import Modal from "./Modal/Modal"
import Nav from "./Nav"
import Text from "./Text"
import TextArea from "./TextArea"
import TextLink from "./TextLink"
import Title from "./Title"

import React from "react"
import ReactDOM from "react-dom"
import Relay from "react-relay"

import { artsyNetworkLayer } from "../Relay/config"
import * as Artsy from "./Artsy"

interface InitOptions {
  user: User
  component: any
  domID: string
  queryConfig: any
}

export function init(options: InitOptions) {
  Relay.injectNetworkLayer(artsyNetworkLayer(options.user))

  const rootRoute = (
    <Artsy.ContextProvider currentUser={options.user}>
      <Relay.RootContainer Component={options.component} route={options.queryConfig} />
    </Artsy.ContextProvider>
  )

  ReactDOM.render(rootRoute, document.getElementById(options.domID))
}

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Gene,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
