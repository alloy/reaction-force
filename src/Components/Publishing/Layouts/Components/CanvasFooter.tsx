import { color, Separator } from "@artsy/palette"
import { CollectionsRailContent } from "Components/CollectionsRail"
import {
  ArticleData,
  DisplayData,
  RelatedArticleCanvasData,
} from "Components/Publishing//Typings"
import { DisplayCanvas } from "Components/Publishing/Display/Canvas"
import { RelatedArticlesCanvas } from "Components/Publishing/RelatedArticles/Canvas/RelatedArticlesCanvas"
import React from "react"
import styled from "styled-components"

export interface CanvasFooterProps {
  display?: DisplayData
  relatedArticles?: RelatedArticleCanvasData[]
  article: ArticleData
  renderTime?: number
  showCollectionsRail?: boolean
  adUnit?: string
  adDimension?: string
}

export const CanvasFooter: React.SFC<CanvasFooterProps> = props => {
  const {
    article,
    display,
    relatedArticles,
    renderTime,
    adDimension,
    adUnit,
  } = props

  return (
    <CanvasFooterContainer>
      {relatedArticles && (
        <RelatedArticlesCanvas
          articles={relatedArticles}
          vertical={article.layout !== "news" && article.vertical}
        />
      )}

      {props.showCollectionsRail && (
        <div>
          <Separator mb={4} />
          <CollectionsRailContent {...props} articleId={article.id} />
        </div>
      )}

      {display && (
        <DisplayContainer hasBorder={relatedArticles ? true : false}>
          <DisplayCanvas
            unit={display.canvas}
            campaign={display}
            article={article}
            renderTime={renderTime}
            adUnit={adUnit}
            adDimension={adDimension}
          />
        </DisplayContainer>
      )}
    </CanvasFooterContainer>
  )
}

export const CanvasFooterContainer = styled.div`
  border-top: 1px solid ${color("black10")};
`

const DisplayContainer = styled.div.attrs<{ hasBorder?: boolean }>({})`
  ${props =>
    props.hasBorder &&
    `
    border-top: 1px solid ${color("black10")}; 
  `};
  padding-bottom: 20px;
`
