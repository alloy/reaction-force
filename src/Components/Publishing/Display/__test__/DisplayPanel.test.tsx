import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Campaign, UnitPanel } from "../../Fixtures/Components"
import { DisplayPanel } from "../DisplayPanel"

it("renders the display panel", () => {
  const displayPanel = renderer.create(<DisplayPanel unit={UnitPanel} campaign={Campaign} />).toJSON()
  expect(displayPanel).toMatchSnapshot()
})
