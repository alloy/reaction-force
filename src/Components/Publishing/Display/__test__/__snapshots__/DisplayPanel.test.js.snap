// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders the display panel with an image 1`] = `
.c0 {
  cursor: pointer;
  text-decoration: none;
  color: black;
  max-width: 360px;
}

.c3 {
  margin-bottom: 15px;
  width: 100%;
  height: 142px;
  background-color: black;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #e5e5e5;
  padding: 20px;
  max-width: 360px;
  box-sizing: border-box;
}

.c1 .c2 {
  background: url(https://de5y2r7wr8mpb.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FYqTtwB7AWqKD95NGItwjJg%252FRachel_Rossin_portrait_2.jpg&width=680&height=284&quality=95) no-repeat center center;
  background-size: cover;
}

.c1 .s1qpl4no-1-file__Image-iTToDH {
  background: url(https://de5y2r7wr8mpb.cloudfront.net?resize_to=fill&src=&width=680&height=284&quality=95) no-repeat center center;
  background-size: cover;
}

.c4 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  line-height: 1.23em;
  margin-bottom: 3px;
}

.c5 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  line-height: 1.53em;
  margin-bottom: 20px;
}

.c5 a {
  color: black;
}

.c6 {
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #e5e5e5;
}

@media (max-width:720px) {
  .c0 {
    margin: 0 auto;
  }
}

@media (max-width:900px) {
  .c1 {
    margin: auto;
  }
}

@media (max-width:720px) {
  .c1 {
    margin: auto;
  }
}

@media (max-width:600px) {
  .c1 {
    margin: auto;
  }
}

<div
  className="c0"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="DisplayPanel__DisplayPanelContainer c1"
  >
    <div
      className="DisplayPanel__Image c2 c3"
    />
    <div>
      <div
        className="c4"
      >
        Euismod Inceptos Quam
      </div>
      <div
        className="c5"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Donec sed odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. <a href='http://artsy.net/articles'>Example Link</a>",
          }
        }
      />
      <div
        className="c6"
      >
        Sponsored by Artsy
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshots renders the display panel with video 1`] = `
.c0 {
  cursor: pointer;
  text-decoration: none;
  color: black;
  max-width: 360px;
}

.c4 {
  margin-bottom: 15px;
  width: 100%;
  height: 142px;
  background-color: black;
  box-sizing: border-box;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.fiUFIk .c3 {
  background: url(https://de5y2r7wr8mpb.cloudfront.net?resize_to=fill&src=&width=680&height=284&quality=95) no-repeat center center;
  background-size: cover;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #e5e5e5;
  padding: 20px;
  max-width: 360px;
  box-sizing: border-box;
}

.c1 .file__Image-s1qpl4no-1 {
  background: url(https://de5y2r7wr8mpb.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FlHEsRROMLasYi9yZtgTR7A%252Fbombay_artsy_panel_640_28_480.mp4&width=680&height=284&quality=95) no-repeat center center;
  background-size: cover;
}

.c1 .c3 {
  background: url(https://de5y2r7wr8mpb.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FYqTtwB7AWqKD95NGItwjJg%252FRachel_Rossin_portrait_2.jpg&width=680&height=284&quality=95) no-repeat center center;
  background-size: cover;
}

.c6 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  line-height: 1.23em;
  margin-bottom: 3px;
}

.c7 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  line-height: 1.53em;
  margin-bottom: 20px;
}

.c7 a {
  color: black;
}

.c8 {
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #e5e5e5;
}

.c2 {
  margin-bottom: 15px;
  width: 100%;
  height: 142px;
  background-color: black;
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 video {
  object-fit: cover;
  object-position: 50%;
  width: 100%;
  height: 100%;
}

.c5 {
  color: white;
  border-top: 13px solid transparent;
  border-bottom: 13px solid transparent;
  border-left: 20px solid white;
  position: absolute;
  bottom: 13px;
  left: 15px;
}

@media (max-width:720px) {
  .c0 {
    margin: 0 auto;
  }
}

@media (max-width:900px) {
  .c1 {
    margin: auto;
  }
}

@media (max-width:720px) {
  .c1 {
    margin: auto;
  }
}

@media (max-width:600px) {
  .c1 {
    margin: auto;
  }
}

<div
  className="c0"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="DisplayPanel__DisplayPanelContainer c1"
  >
    <div
      className="VideoContainer c2"
    >
      <div
        className="VideoContainer__VideoCover c3 c4"
      />
      <video
        className="VideoContainer__video"
        controls={false}
        playsInline={true}
        src="https://artsy-media-uploads.s3.amazonaws.com/lHEsRROMLasYi9yZtgTR7A%2Fbombay_artsy_panel_640_28_480.mp4"
      />
      <div
        className="c5"
      />
    </div>
    <div>
      <div
        className="c6"
      >
        Euismod Inceptos Quam
      </div>
      <div
        className="c7"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Donec sed odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. <a href='http://artsy.net/articles'>Example Link</a>",
          }
        }
      />
      <div
        className="c8"
      >
        Sponsored by Artsy
      </div>
    </div>
  </div>
</div>
`;
